golem-service-base/src/error/worker_executor.rs

```diff
@@ -876,3 +910,44 @@ impl Display for WorkerOutOfMemory {
 }
 
 impl Error for WorkerOutOfMemory {}
+
+#[derive(
+    Debug, Clone, PartialOrd, PartialEq, Eq, Hash, Serialize, Deserialize, Encode, Decode, IntoValue,
+)]
+pub enum InterruptKind {
+    Interrupt,
+    Restart,
+    Suspend,
+    Jump,
+}
+
+impl Display for InterruptKind {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            InterruptKind::Interrupt => write!(f, "Interrupted via the Golem API"),
+            InterruptKind::Restart => write!(f, "Simulated crash via the Golem API"),
+            InterruptKind::Suspend => write!(f, "Suspended"),
+            InterruptKind::Jump => write!(f, "Jumping back in time"),
+        }
+    }
+}
+
+impl Error for InterruptKind {}
+
+#[cfg(feature = "worker-executor")]
```

Comment:
just to avoid pulling in wasmtime when only compiling a single service in ci that doesn't need it.

This will cause rebuilds when switching between the two (workspace build and single service build), so there is an argument for just alway building it.